src/StatusQ/Buttons/StatusButton.qml

```diff
@@ -0,0 +1,165 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+
+Rectangle {
+    id: statusButton
+
+    enum Size {
+        Small,
+        Large
+    }
+
+    enum Type {
+        Normal,
+        Danger,
+        WithoutBackground,
+        DangerWithoutBackground
+    }
+
+    property bool loading: false
+
+    property int size: StatusButton.Size.Large
+    property int type: StatusButton.Type.Normal
+
+    property alias text: label.text
+    property alias font: label.font
+
+    property real leftPadding: size === StatusButton.Size.Large ? 24 : 12 // by design
+    property real rightPadding: size === StatusButton.Size.Large ? 24 : 12 // by design
+    property real topPadding: size === StatusButton.Size.Large ? 11 : 10 // by design
+    property real bottomPadding: size === StatusButton.Size.Large ? 11 : 10 // by design
+
+    signal pressed(var mouse)
+
+
+
+    /// Implementation
```

Comment:
This comment we can remove

Reply:
Its annotation where `public` api

Reply:
Is that a QtCreator thing? If this is a standard way of marking public APIs in QML, we should make sure we do it in all of our components.

Reply:
No, but if you check Qt QML sources i can find something similar.